extensions/sql-database-projects/src/models/deploy/deployService.ts

```diff
@@ -0,0 +1,389 @@
+import * as utils from '../../common/utils';
+let fse = require('fs-extra');
+let path = require('path');
+import * as childProcess from 'child_process';
+import * as vscode from 'vscode';
+import * as os from 'os';
+import { ConnectionResult } from 'azdata';
+
+interface ValidationResult {
+	errorMessage: string;
+	validated: boolean
+}
+export class DeployService {
+
+	constructor(private _outputChannel: vscode.OutputChannel) {
+	}
+
+	private createConnectionStringTemplate(runtime: string | undefined): string {
+		switch (runtime?.toLocaleLowerCase()) {
+			case 'dotnet':
+				return constants.defaultConnectionStringTemplate;
+				break;
+			// TODO: add connection strings for other languages
+			default:
+				break;
+		}
+		return '';
+	}
+
+	private findAppRuntime(profile: IDeployProfile, appSettingContent: any): string | undefined {
+		switch (profile.appSettingType) {
+			case AppSettingType.AzureFunction:
+				return <string>appSettingContent?.Values['FUNCTIONS_WORKER_RUNTIME'];
+			default:
+		}
+		return undefined;
+	}
+
+	public async updateAppSettings(profile: IDeployProfile): Promise<void> {
+		// Update app settings
+		//
+		if (profile.appSettingFile) {
+			this.logToOutput(`Updating app setting: ${profile.appSettingFile}`);
+
+			let content = JSON.parse(fse.readFileSync(profile.appSettingFile, 'utf8'));
+			if (content && content.Values) {
+				let connectionString: string | undefined = '';
+				if (profile.localDbSetting) {
+					// Find the runtime and generate the connection string for the runtime
+					//
+					const runtime = this.findAppRuntime(profile, content);
+					let connectionStringTemplate = this.createConnectionStringTemplate(runtime);
+					connectionString = connectionStringTemplate?.
+						replace('{#SERVER#}', profile?.localDbSetting?.serverName || '').
+						replace('{#PORT#}', profile?.localDbSetting?.port?.toString() || '').
+						replace('{#USER#}', profile?.localDbSetting?.userName || '').
+						replace('{#SA_PASSWORD#}', profile?.localDbSetting?.password || '').
+						replace('{#DATABASE#}', profile?.localDbSetting?.dbName || '');
+				} else if (profile.deploySettings?.connectionUri) {
+					connectionString = await this.getConnectionString(profile.deploySettings?.connectionUri);
+				}
+
+				if (connectionString && profile.envVariableName) {
+					content.Values[profile.envVariableName] = connectionString;
+					await fse.writeFileSync(profile.appSettingFile, JSON.stringify(content, undefined, 4));
+					this.logToOutput(`app setting '${profile.appSettingFile}' has been updated. env variable name: ${profile.envVariableName} connection String: ${connectionString}`);
+
+				} else {
+					this.logToOutput(constants.deployAppSettingUpdateFailed(profile.appSettingFile));
+				}
+			}
+		}
+	}
+
+	public async deploy(profile: IDeployProfile, project: Project): Promise<string | undefined> {
+
+		return await this.executeTask(constants.deployDbTaskName, async () => {
+			if (profile.localDbSetting) {
```

Comment:
Since there's no alternate behavior, may as well

```
if (!profile.localDbSetting) {
    return undefined;
}
[everything else]
```
and promote the bulk of the logic a level.